inference/models/transformers/transformers.py

```diff
@@ -0,0 +1,261 @@
+
+    def postprocess(
+        self,
+        predictions: Tuple[str],
+        preprocess_return_metadata: PreprocessReturnMetadata,
+        **kwargs,
+    ) -> LMMInferenceResponse:
+        text = predictions[0]
+        image_dims = preprocess_return_metadata["image_dims"]
+        response = LMMInferenceResponse(
+            response=text,
+            image=InferenceResponseImage(width=image_dims[0], height=image_dims[1]),
+        )
+        return [response]
+
+    def predict(self, image_in: Image.Image, prompt="", history=None, **kwargs):
+        model_inputs = self.processor(
+            text=prompt, images=image_in, return_tensors="pt"
+        ).to(self.model.device)
+        input_len = model_inputs["input_ids"].shape[-1]
+
+        with torch.inference_mode():
+            generation = self.model.generate(
+                **model_inputs, max_new_tokens=100, do_sample=False
+            )
+            generation = generation[0]
+            if self.generation_includes_input:
+                generation = generation[input_len:]
+            decoded = self.processor.decode(generation, skip_special_tokens=True)
+
+        return (decoded,)
+
+    def get_infer_bucket_file_list(self) -> list:
+        """Get the list of required files for inference.
+
+        Returns:
+            list: A list of required files for inference, e.g., ["model.pt"].
+        """
+        return [
+            "config.json",
+            "special_tokens_map.json",
+            "generation_config.json",
+            "model.safetensors.index.json",
+            "tokenizer.json",
+            re.compile(r"model-\d{5}-of-\d{5}\.safetensors"),
+            "preprocessor_config.json",
+            "tokenizer_config.json",
+        ]
+
+    def download_model_artifacts_from_roboflow_api(self) -> None:
+        api_data = get_roboflow_model_data(
+            api_key=self.api_key,
+            model_id=self.endpoint,
+            endpoint_type=ModelEndpointType.ORT,
+            device_id=self.device_id,
+        )
+        if "weights" not in api_data["ort"]:
+            raise ModelArtefactError(
+                f"`weights` key not available in Roboflow API response while downloading model weights."
+            )
+        for weights_url in api_data["ort"]["weights"].values():
+            t1 = perf_counter()
+            filename = weights_url.split("?")[0].split("/")[-1]
+            if filename.endswith(".npz"):
+                continue
+            model_weights_response = get_from_url(weights_url, json_response=False)
+            save_bytes_in_cache(
+                content=model_weights_response.content,
+                file=filename,
+                model_id=self.endpoint,
+            )
+            if perf_counter() - t1 > 120:
+                logger.debug(
+                    "Weights download took longer than 120 seconds, refreshing API request"
+                )
+                api_data = get_roboflow_model_data(
```

Comment:
it seems we do nothing with this, is this required?

Reply:
this is refreshing the download links in case the download takes too long, which ensures all download links are valid